### ISSUE TITLE
Auto download model.

### ISSUE THREAD

**Issue by Vozf** [GantMan/nsfw_model#80]
It would be convenient for model to download automatically without additional manual download actions

**Comment by Vozf** [comment]
Where is the file from the tutorial `nsfw_mobilenet2.224x224.h5`? It is not present in linked archive

**Comment by Vozf** [comment]
There is no any files by that name in the archive. But a lot of unrelated files. Trying all the h5 files ends with a python exception
![image](https://user-images.githubusercontent.com/22998537/102995254-b9ed9000-4531-11eb-9efd-5f427ed230ff.png)

**Comment by Vozf** [comment]
Eh, this is one of the most raw projects I've seen with that number of stars. Basically nothing works here. You've got to deduce which model to use from the endless archive. Even if you guess correct, it won't work because you need to modify mode loading function. 
Even if you get the model loaded prediction on list of paths is not working because author decided to use `isdir` to chech for dir which throws exception if list is fed. 
![image](https://user-images.githubusercontent.com/22998537/102997082-48afdc00-4535-11eb-8791-59a1066bc2cb.png)
Quite interesting how there is no ready to use project to classify nsfw on the whole github.
For anyone wondering this is what I got working
Download the https://github.com/GantMan/nsfw_model/releases/download/1.2.0/mobilenet_v2_140_224.1.zip and unpack it. The model is inside `saved_model.h5`
To load it don't use load_model from project, because it will throw keraslayer exception. To load model use this
```
    import tensorflow_hub

    model = tf.keras.models.load_model(
        "saved_model.h5", custom_objects={"KerasLayer": tensorflow_hub.KerasLayer}
    )
```
To predict you can use classify but you can't use list of paths. Only single prediction or folder will work

**Comment by GantMan** [comment]
Fair assessment.   

To be honest I'm not terribly familiar with Python.  I write just enough to serve my AI/ML needs and lately I've been kicking butt over in JavaScript where the fruits of this project have gone far and wide (https://nsfwjs.com/ and https://github.com/infinitered/nsfwjs)

This project used to run back with TF 1.0 and since then it's had a few people show up and help.   I could really use the help in getting it in shape, bc I've personally been focused with other things for a couple years.  The issue is that I haven't touched the Python side in a long time.


If you're willing to do the open source work to get the model working like you want, I'd make you a collaborator for the project.

**Comment by MassoudHasanpour** [comment]
Hi,
I have the same problem with loading the trained model. I tried all of the above solutions but no one worked. When I want to load **"saved_model.h5"**, I face the following error:
**_WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually._**

Can anyone help me please or let me know if there is a way to access the trained model?

**Comment by Dicklesworthstone** [comment]
I agree, it was quite annoying to get it to work in Python. Here is my solution:

First, download the latest model files from https://github.com/GantMan/nsfw_model/releases/download/1.2.0/mobilenet_v2_140_224.1.zip and extract them somewhere. Set nsfw_model_path to this directory. Then use this function:

def get_nsfw_scores_func(nsfw_model_path, input_image_path):
    tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
    print('Loading NSFW Tensorflow model...')
    nsfw_model = keras.models.load_model(nsfw_model_path, compile=False, custom_objects={'KerasLayer': hub.KerasLayer})
    input_image = keras.preprocessing.image.load_img(input_image_path, target_size=(224,224))
    input_image = keras.preprocessing.image.img_to_array(input_image)
    input_image /= 255
    input_image = input_image[np.newaxis is None,:,:,:]
    model_preds = nsfw_model.predict(input_image)[0]
    print('Generating NSFW predictions from input image...')
    model_preds_dict = {
        'drawings': model_preds[0],
        'hentai': model_preds[1],
        'neutral': model_preds[2],
        'porn': model_preds[3],
        'sexy': model_preds[4]}
    return model_preds_dict

**Comment by tovganesh** [comment]
> Fair assessment.
> 
> To be honest I'm not terribly familiar with Python. I write just enough to serve my AI/ML needs and lately I've been kicking butt over in JavaScript where the fruits of this project have gone far and wide (https://nsfwjs.com/ and https://github.com/infinitered/nsfwjs)
> 
> This project used to run back with TF 1.0 and since then it's had a few people show up and help. I could really use the help in getting it in shape, bc I've personally been focused with other things for a couple years. The issue is that I haven't touched the Python side in a long time.
> 
> If you're willing to do the open source work to get the model working like you want, I'd make you a collaborator for the project.

I also have limited know-how of Tf, but I am willing to help here. Please let me know how can we collaborate on improving this further.

**Comment by SourojitGhosh** [comment]
> Where is the file from the tutorial `nsfw_mobilenet2.224x224.h5`? It is not present in linked archive

!wget https://s3.amazonaws.com/ir_public/nsfwjscdn/nsfw_mobilenet2.224x224.h5
nsfw_model = predict.load_model('./nsfw_mobilenet2.224x224.h5')
